packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -372,6 +382,128 @@ const Account = Backbone.Model.extend(
       });
     },
 
+    _fetchAccountProfileResult: null,
+
+    /**
+     * Fetches the account profile from the GET /account/profile on the auth server and caches the
+     * result locally in _accountProfileFetchPromise. This also updates the account model
+     * with the new properties (minus any verificationMethod, verificationReason values)
+     *
+     * @returns {Promise}
+     */
+    async fetchAccountProfile() {
+      if (this._fetchAccountProfileResult) {
+        return this._fetchAccountProfileResult;
+      }
+
+      const result = await this.accountProfile();
+
+      // Currently, we only populate the `ecosystemAnonId`, however we could examine populating
+      // other account values.
+      if (result.ecosystemAnonId) {
+        this.set('ecosystemAnonId', result.ecosystemAnonId);
+      }
+
+      this._fetchAccountProfileResult = result;
+
+      return result;
+    },
+
+    /**
+     * Attempt to generate and set the ecosystem anon if it isn't already set. At a high level
+     * this function attempts to generate the id in the following manner:
+     *
+     * 1. Use kB if it is specified
+     * 2. Attempt to fetch keys if valid authentication token exists to do so
+     * 3. Use user's password to fetch keys
+     *
+     *
+     * @param {Object} [options={}]
+     *   @param {String} [options.kB] Optional The user's kB, ff specified takes priority.
+     *   @param {String} [options.password] Optional The user's password
+     * @returns {Promise}
+     */
+    async generateEcosystemAnonIdIfRequired(options = {}) {
+      try {
+        let ecosystemAnonId = this.get('ecosystemAnonId');
+        if (ecosystemAnonId) {
+          return;
+        }
+
+        // If no data pipeline keys are available, we can't generate the ecosystemAnonId, abort
+        if (this._ecosystemAnonIdPublicKeys.length <= 0) {
+          return;
+        }
+
+        // Pick a random public key
+        const randomEcosystemAnonIdPublicKey = this._ecosystemAnonIdPublicKeys[
+          Math.floor(Math.random() * this._ecosystemAnonIdPublicKeys.length)
+        ];
+
+        //
+        if (options.kB) {
+          ecosystemAnonId = await ecosystemAnonKeys.generateEcosystemAnonID(
+            options.kB,
+            randomEcosystemAnonIdPublicKey
+          );
+        } else if (this.canFetchKeys()) {
+          const keys = await this.accountKeys();
+          ecosystemAnonId = await ecosystemAnonKeys.generateEcosystemAnonID(
+            keys.kB,
+            randomEcosystemAnonIdPublicKey
+          );
+        } else if (options.password) {
+          // TODO: Using sessionReauth to get keys could potentially pose some issues if the user
+          // is attempting to come from an unverified session. FxA only allows fetching keys from
+          // sessions that are verified (ex. performed some email verification loop).
```

Comment:
This seems like an important TODO. How should we handle it? Maybe file a followup issue to discuss?

Reply:
Filed an issue here: https://github.com/mozilla/fxa/issues/6018